test/integration-tests/V3ToV4Migration.test.ts

```diff
@@ -0,0 +1,856 @@
+          v: v,
+          r: r,
+          s: s,
+        }
+
+        const encodedErc721PermitCall = encodeERC721Permit(erc721PermitParams)
+
+        planner.addCommand(CommandType.ERC721_PERMIT, [encodedErc721PermitCall])
+
+        let position = await v3NFTPositionManager.positions(tokenIdv3)
+        let liquidity = position.liquidity
+
+        // set the deadline to 0
+        const decreaseParams = { tokenId: tokenIdv3, liquidity: liquidity, amount0Min: 0, amount1Min: 0, deadline: '0' }
+
+        const encodedDecreaseCall = encodeDecreaseLiquidity(decreaseParams)
+
+        planner.addCommand(CommandType.V3_POSITION_MANAGER_CALL, [encodedDecreaseCall])
+
+        // call to decrease liquidity fails since the deadline is set to 0
+        await expect(
+          executeRouter(planner, bob, router, wethContract, daiContract, usdcContract)
+        ).to.be.revertedWithCustomError(router, 'ExecutionFailed')
+      })
+
+      it('cannot call decrease liquidity if not authorized', async () => {
+        // bob creates a signature for the router to spend the token
+        const { v, r, s } = await getPermitNFTSignature(bob, v3NFTPositionManager, router.address, tokenIdv3, MAX_UINT)
+        const erc721PermitParams = {
+          spender: router.address,
+          tokenId: tokenIdv3,
+          deadline: MAX_UINT,
+          v: v,
+          r: r,
+          s: s,
+        }
+
+        const encodedErc721PermitCall = encodeERC721Permit(erc721PermitParams)
+
+        planner.addCommand(CommandType.ERC721_PERMIT, [encodedErc721PermitCall])
+
+        await executeRouter(planner, bob, router, wethContract, daiContract, usdcContract)
+
+        planner = new RoutePlanner()
+
+        // transfer the token to eve
+        await v3NFTPositionManager.transferFrom(bob.address, eve.address, tokenIdv3)
+
+        let position = await v3NFTPositionManager.positions(tokenIdv3)
+        let liquidity = position.liquidity
+
+        const decreaseParams = {
+          tokenId: tokenIdv3,
+          liquidity: liquidity,
+          amount0Min: 0,
+          amount1Min: 0,
+          deadline: MAX_UINT,
+        }
+
+        const encodedDecreaseCall = encodeDecreaseLiquidity(decreaseParams)
+
+        planner.addCommand(CommandType.V3_POSITION_MANAGER_CALL, [encodedDecreaseCall])
+
+        // bob is trying to use the token that is now owned by eve. he is not authorized to do so
+        await expect(
+          executeRouter(planner, bob, router, wethContract, daiContract, usdcContract)
+        ).to.be.revertedWithCustomError(router, 'NotAuthorizedForToken')
+      })
+
+      it('bob is permitted over the nft so he can call decrease even though he is not the owner', async () => {
+        // transfer the token to eve
+        await v3NFTPositionManager.transferFrom(bob.address, eve.address, tokenIdv3)
+
+        // eve permits bob to spend the token
+        await v3NFTPositionManager.connect(eve).setApprovalForAll(bob.address, true)
```

Comment:
would be great to have another test just like this that just approves the 1 specific token (not approval for all)

Reply:
so the opposite? the router is approved for all and bob is permitted for the 1 specific token?

Reply:
still do PERMIT command so that doesnt need to change, just Bob is permitted for only 1 token